### PR TITLE
DDCE-5134 - Fix migration jobs trigger logic

### PR DESCRIPTION
**Pull request by frikit**
Logic is implemented in app/uk/gov/hmrc/bindingtariffclassification/migrations/MigrationRunner.scala and its Spec, rest of it is scalafmt and couple deps upgrades

### CODE REVIEW

File: test/uk/gov/hmrc/bindingtariffclassification/model/CaseRequestSpec.scala [hunk 1]
```diff
@@ -35,7 +35,7 @@ class CaseRequestSpec extends BaseSpec with MockitoSugar {
       when(application.asBTI).thenReturn(CaseData.createBasicBTIApplication)
       val c = NewCaseRequest(application, attachments).toCase("reference")
       c.status               shouldBe CaseStatus.NEW
-      c.createdDate          should roughlyBe(Instant.now())
+      c.createdDate            should roughlyBe(Instant.now())
```
Comment (gastonracui) [inline test/uk/gov/hmrc/bindingtariffclassification/model/CaseRequestSpec.scala:38]: this seems odd

Comment (frikit) [reply test/uk/gov/hmrc/bindingtariffclassification/model/CaseRequestSpec.scala:38]: method `roughlyBe` or what is odd?

File: test/uk/gov/hmrc/bindingtariffclassification/migrations/MigrationRunnerTest.scala [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+import java.time.ZonedDateTime.now
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future.{failed, successful}
+
+class MigrationRunnerTest extends BaseSpec {
+
+  class Test {
+    private def jobName                              = "MigrationJobTest"
+    val migrationJob: MigrationJob                   = mock[MigrationJob]
```
Comment (gastonracui) [inline test/uk/gov/hmrc/bindingtariffclassification/migrations/MigrationRunnerTest.scala:34]: the base spec extends MockitoSugar, do you think will be too much work to remove it in line with the other tickets we are playing?

File: app/uk/gov/hmrc/bindingtariffclassification/repository/MigrationLockRepository.scala [hunk 3]
```diff
@@ -82,6 +84,9 @@ class MigrationLockMongoRepository @Inject() (mongoComponent: MongoComponent)(im
   override def deleteAll(): Future[Unit] =
     collection.deleteMany(empty()).toFuture().flatMap(_ => Future.unit)
 
+  override def findOne(name: String): Future[Option[JobRunEvent]] =
+    collection.find(equal("name", name)).headOption()
+
   /** Tells if this error is due to a write on a secondary node. */
   private def isNotAPrimaryError(code: Int): Boolean = Some(code).exists {
     case 10058 | 10107 | 13435 | 13436 => true
```
Comment (gastonracui) [inline app/uk/gov/hmrc/bindingtariffclassification/repository/MigrationLockRepository.scala:92]: do we know the name of this error codes?

**Review by gastonracui — APPROVED** [review]

**Pull request merged**